### PR TITLE
[release/6.0-rc1] HTTP/3: Response drain timeout

### PR DESCRIPTION
**Pull request by JamesNK**
Backport of https://github.com/dotnet/aspnetcore/pull/35322 to release/6.0-rc1

## Customer Impact

Response drain timeout is a security feature. It hardens Kestrel against certain attacks. This feature exists in HTTP/1.1 and HTTP/2. This PR adds it to HTTP/3.

## Testing

Unit tests and functional tests.

## Risk

Low. This feature only impacts HTTP/3. HTTP/3 is a preview feature in .NET 6.

### CODE REVIEW

**Review by Tratcher — APPROVED** [review]

**Review by adityamandaleeka — APPROVED** [review]
@davidfowl

**Comment by JamesNK** [comment]
Contact me before merging this. One of the other PRs has a unit test that needs to be updated. Will make changes here after it is merged.

**Comment by davidfowl** [comment]
20 files changed... Let me take a deeper look.

**Comment by JamesNK** [comment]
> Contact me before merging this. One of the other PRs has a unit test that needs to be updated. Will make changes here after it is merged.

Done

**Review by halter73 — APPROVED** [review]

**Comment by JamesNK** [comment]
I don't have The Power to click merge.

@dougbu, or someone else with the right permissions, could you merge it?

**Comment by dougbu** [comment]
@davidfowl where / when was this `servicing-approved`❔ Just double-checking because this isn't in the `Servicing Status` project.

**Comment by adityamandaleeka** [comment]
@dougbu This is for 6.0 RC1. I don't believe those are being tracked in that project.

**Comment by dougbu** [comment]
> I don't believe those are being tracked in that project.

The project is one of the ways we track all conversations w/ Tactics about issues / PRs that require approval. If it's not necessary for the current RC1 state, :ok: but we'll definitely need to use the project in RC2. @Pilchie is my understanding correct❔

**Comment by Pilchie** [comment]
I'll admit that I haven't been paying as much attention to that project since https://tacticsview.azurewebsites.net/ was made.

**Comment by dougbu** [comment]
> I'll admit that I haven't been paying as much attention to that project since https://tacticsview.azurewebsites.net/ was made.

Should we stop using the project, just delete it, &hellip;❔

**Comment by dougbu** [comment]
> @davidfowl where / when was this servicing-approved❔

I don't think my question was answered. I'm here to merge if this had the right 👀 on it.

**Comment by dougbu** [comment]
> Should we stop using the project, just delete it, …❔

/fyi we have Fabric bot rules that do "stuff" with the `Servicing Status` project but no rules for 'release/6.0-rc1'. I lean toward reducing our rule complexity and ignoring or deleting the project if that's allowed for existing servicing branches.

**Comment by Pilchie** [comment]
We can go ahead and merge this based on David's approval on Friday (he had delegated M2 approval then).  In the future we should try to get things merged within 24-hours of approval though, as the bar does go up.

Regarding the servicing project, I'm fine with deleting it and simplifying the bot's config if no one is getting value from the project.  Is anyone?  Maybe we should ask at a DoI meeting before just deleting?

**Comment by dougbu** [comment]
> Maybe we should ask at a DoI meeting before just deleting?

Good place to ask but that might miss the people who care. How 'bout an email❔

**Pull request merged**
